### ISSUE TITLE
[a11y]: MODAL - Loss of content is observed when page is magnified to 400%

### ISSUE THREAD

**Issue by LyndseyBu** [carbon-design-system/carbon#20259]
### Package

@carbon/react

### Browser

Firefox

### Operating System

MacOS

### Package version

1.81.0

### React version

17.0.2

### Automated testing tool and ruleset

AVT2 manual testing by AVT team

### Description

Issue description: Loss of content is observed when page is magnified to 400%

using storybook as an example : https://react.carbondesignsystem.com/iframe.html?globals=&args=&id=components-modal--danger-modal&viewMode=story

zoom in to 400%

<img width="1724" height="894" alt="Image" src="https://github.com/user-attachments/assets/c2a231cf-b29e-4eeb-acc7-3eddb60b8a00" />

### WCAG 2.1 Violation

WCAG Mapping: 1.4.10 Reflow

### Reproduction/example

 https://react.carbondesignsystem.com/iframe.html?globals=&args=&id=components-modal--danger-modal&viewMode=story

### Steps to reproduce

https://react.carbondesignsystem.com/iframe.html?globals=&args=&id=components-modal--danger-modal&viewMode=story
zoom browser to 400%

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hi @LyndseyBu, thanks for opening this up! My understanding of the success criterion is that when magnified, text must reflow normally so users only have to scroll in one direction - the direction of reading. 

Content within the modal is reflowing in this way and the user can scroll to read the remaining content. 

https://github.com/user-attachments/assets/1d02bd95-e19b-4b6c-adce-e7bbe43de535

The second part of this is that at certain sizes the fade/overlay obscures the text, like what's shown in your screenshot. I'm honestly not sure how much we need to handle in this situation. If you put the modal into a hugely magnified small viewport there's all sorts of problems. 

https://github.com/user-attachments/assets/eeb9eacb-9ece-4376-9461-68b32a95fe22

I know that's a contrived example but I'm honestly just not sure where that line is of what must be fixed vs. not and/or which of these are valid enough use cases that we need to address. 

Looping in the IBM Accessibility team is probably the next step. @mbgower I'm curious your thoughts.

Depending on the outcome, this will probably need some design spec work to know how we should handle this situation(s).

**Comment by LyndseyBu** [comment]
For accessibility guidelines you should be able to scale up to 400%, in this case the overlay seems unnecessary and is blocking the the actual content. As you can see from our screenshots below we have a longer title which makes the content area even smaller and is almost completely take over by the overlay.

as per the guidance https://www.ibm.com/able/requirements/requirements/?version=v7_3#1_4_10 

This issue was raised with us by our AVT test team and tested with these dimensions as outlined `1280x1024 pixel dimensions by zooming the content to 400%`.

https://github.ibm.com/org-ids/roadmap/issues/28736#issuecomment-127830837


<img width="750" height="605" alt="Image" src="https://github.com/user-attachments/assets/4c34f31c-cb9b-4766-a6d8-0686c65249c5" />

<img width="743" height="624" alt="Image" src="https://github.com/user-attachments/assets/e13bdc17-7508-4aba-bfc1-cd8219553eb3" />

**Comment by tay1orjones** [comment]
@LyndseyBu Yes, something the IBM Accessibility team pointed out to us was

> Content can be presented without loss of information or functionality, and without requiring scrolling in two dimensions for:
> * Vertical scrolling content at a width equivalent to 320 [CSS pixels](https://www.w3.org/TR/WCAG22/#dfn-css-pixels);
> * Horizontal scrolling content at a height equivalent to 256 [CSS pixels](https://www.w3.org/TR/WCAG22/#dfn-css-pixels).
> 
> Except for parts of the content which require two-dimensional layout for usage or meaning.

Right now that is working fine, reflow is passing. The overlay/fade problem is more specifically failing [lack of contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html), but these reflow parameters above inform to what extend we need to fix the fade issue.

A fix for this would target [`1280px` viewport width at 100% zoom](https://www.w3.org/TR/WCAG22/#reflow:~:text=320%20CSS%20pixels%20is%20equivalent%20to%20a%20starting%20viewport%20width%20of%201280%20CSS%20pixels%20wide%20at%20400%25%20zoom.%20For%20web%20content%20which%20is%20designed%20to%20scroll%20horizontally%20(e.g.%2C%20with%20vertical%20text)%2C%20256%20CSS%20pixels%20is%20equivalent%20to%20a%20starting%20viewport%20height%20of%201024%20CSS%20pixels%20at%20400%25%20zoom.) all the way through to 400% zoom. At 400% it is effectively `320px` width as specified in the reflow link above. For this what needs to be done here is:

# Todo

- [ ] Remove the fade when the modal body container height is too short and obscures too much text. This will prevent the loss of contrast as you've reported. 
- [ ] With the fade removed we can refactor the code delivered in https://github.com/carbon-design-system/carbon/pull/19534 to just call `scrollIntoView()` every time focus is moved within the body.

**Comment by tay1orjones** [comment]
Also, you might double check your modal header styling for overrides. In the default styling it has a maximum height limit so the body text container isn't collapsed beyond a few lines of text. In your screenshot it looks like it's going down to a single line of text somehow. Here's an example of what I mean and a preview of what the end result could look like for removing the fade.

https://github.com/user-attachments/assets/c19ed033-50b9-4e1e-a285-9b4a8504306b

**Comment by mbgower** [comment]
As mentioned on the CAG call, there is at least one other tactic that can be used:
1. Only expand any heading to double its original size. The requirement of [Resize Text](https://www.w3.org/WAI/WCAG22/Understanding/resize-text.html) is that text can be increased to 200%, so if you start with 24pt headings, you need to get to 48pt. But the 400% mentioned in Reflow is _not_ a requirement that all text needs to keep increasing in size as the page zooms. It is allowable, and desirable in cases such as this, that a heading not take over the entirety of a view. It's the smaller body text that we want to keep on increasing the size of, where possible.
